pkg/plugins/vz/vz.go

```diff
@@ -0,0 +1,442 @@
+		Message: "Snapshot deleted successfully",
+	}, nil
+}
+
+// ApplySnapshot implements the ApplySnapshot RPC
+func (p *VzPlugin) ApplySnapshot(ctx context.Context, req *plugins.ApplySnapshotRequest) (*plugins.ApplySnapshotResponse, error) {
+	instance, exists := p.instances[req.InstanceId]
+	if !exists {
+		return &plugins.ApplySnapshotResponse{
+			Success: false,
+			Message: "VM not found",
+		}, nil
+	}
+
+	// Create base driver for snapshot application
+	baseDriver := &driver.BaseDriver{
+		Instance: &store.Instance{
+			Dir:    framework.GetInstanceDir(req.InstanceId),
+			Config: &instance.Config.LimaYAML,
+		},
+	}
+
+	if err := vz.ApplySnapshot(baseDriver, req.Tag); err != nil {
+		return &plugins.ApplySnapshotResponse{
+			Success: false,
+			Message: fmt.Sprintf("failed to apply snapshot: %v", err),
+		}, nil
+	}
+
+	return &plugins.ApplySnapshotResponse{
+		Success: true,
+		Message: "Snapshot applied successfully",
+	}, nil
+}
+
+// ListSnapshots implements the ListSnapshots RPC
+func (p *VzPlugin) ListSnapshots(ctx context.Context, req *plugins.ListSnapshotsRequest) (*plugins.ListSnapshotsResponse, error) {
+	instance, exists := p.instances[req.InstanceId]
+	if !exists {
+		return &plugins.ListSnapshotsResponse{
+			Success: false,
+			Message: "VM not found",
+		}, nil
+	}
+
+	// Create base driver for snapshot listing
+	baseDriver := &driver.BaseDriver{
+		Instance: &store.Instance{
+			Dir:    framework.GetInstanceDir(req.InstanceId),
+			Config: &instance.Config.LimaYAML,
+		},
+	}
+
+	snapshots, err := vz.ListSnapshots(baseDriver)
+	if err != nil {
+		return &plugins.ListSnapshotsResponse{
+			Success: false,
+			Message: fmt.Sprintf("failed to list snapshots: %v", err),
+		}, nil
+	}
+
+	return &plugins.ListSnapshotsResponse{
+		Success: true,
+		Message: "Snapshots listed successfully",
+		Snapshots: snapshots,
+	}, nil
+}
+
+// ChangeDisplayPassword implements the ChangeDisplayPassword RPC
+func (p *VzPlugin) ChangeDisplayPassword(ctx context.Context, req *plugins.ChangeDisplayPasswordRequest) (*plugins.ChangeDisplayPasswordResponse, error) {
+	instance, exists := p.instances[req.InstanceId]
+	if !exists {
+		return &plugins.ChangeDisplayPasswordResponse{
+			Success: false,
+			Message: "VM not found",
+		}, nil
+	}
+
+	// VZ driver does not support display password changes
+	return &plugins.ChangeDisplayPasswordResponse{
+		Success: false,
+		Message: "Display password changes are not supported by the VZ driver",
+	}, nil
+}
+
+// GetDisplayConnection implements the GetDisplayConnection RPC
+func (p *VzPlugin) GetDisplayConnection(ctx context.Context, req *plugins.GetDisplayConnectionRequest) (*plugins.GetDisplayConnectionResponse, error) {
+	instance, exists := p.instances[req.InstanceId]
+	if !exists {
+		return &plugins.GetDisplayConnectionResponse{
+			Success: false,
+			Message: "VM not found",
+		}, nil
+	}
+
+	// VZ driver does not support display connections
+	return &plugins.GetDisplayConnectionResponse{
```

Comment:
Probably gRPC's "UNIMPLEMENTED" error can be used
https://grpc.io/docs/guides/status-codes/